ansible/tasks/nginx/main.yml

```diff
@@ -0,0 +1,52 @@
+---
+- apt_key:
+    url: https://nginx.org/keys/nginx_signing.key
+    state: present
+
+- apt_repository:
+    repo: deb http://nginx.org/packages/debian/ stretch nginx
+    state: present
+
+- apt_repository:
+    repo: deb-src http://nginx.org/packages/debian/ stretch nginx
+    state: present
+
+- name: Update apt repositories
+  apt:
+    update_cache: yes
+
+- name: Install nginx from nginx.org
+  apt:
+    name={{ item }}
+  with_items:
+    - nginx
+
+- name: Setup sites-enabled/hashbang-http
+  register: nginxconf
+  copy:
+    dest: /etc/nginx/sites-enabled/hashbang-http
+    content: |
+      # Hashbang server configuration
+      # ref: https://gist.github.com/RyanSquared/e59c7e274abef06e63b0e47e46997b00
+      #
+      server {
+      	listen 80;
+      	listen [::]:80;
+      
+      	server_name "~^(?<user>[a-zA-Z0-9]+)\.(?<server>\w{2}\d{1})\.hashbang\.sh$";
+      
+      	location / {
+      		proxy_pass http://unix:/home/$user/.nginx.sock;
+                proxy_set_header X-Real-IP       $remote_addr;
+                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+      	}
+      }
+
+- name: Remove default nginx site
+  file:
+   state: absent
+   path: /etc/nginx/sites-enabled/default
```

Comment:
Should we also remove /etc/nginx/sites-available/default ?

Reply:
/etc/nginx/sites-available/default is src of this symlink. there is no point to remove the original copy

Reply:
Agreed